### PR TITLE
Fix serialization issue in the config file

### LINKED ISSUE #1321: [Bug]: When using dab init with a password with special characters, it escapes them and the DB cannot login
### What happened?

I run the following:
dab init --database-type "mssql" --connection-string "Server=localhost,1439;Database=dabDB;User ID=btspendo;Password=5E5C2k*M$o&7mYU66!49#tZ1&CN9*X2BUj&PCGp7CD@%y%p1x1;TrustServerCertificate=true" --host-mode "Development"
When I look in my dab-config.json file, the password is now:
5E5C2k*M**\u00267**mYU66!49#tZ1**\u0026**CN9*X2BUj**\u0026**PCGp7CD@%y%p1x1
and I cannot login to the database because the &'s are being escaped.

### Version

Microsoft.DataApiBuilder 0.5.34

### What database are you using?

Azure SQL

### What hosting model are you using?

Local (including CLI), Custom Docker host

### Which API approach are you accessing DAB through?

_No response_

### Relevant log output

```Text
fail: Azure.DataApiBuilder.Service.Startup[0]
      Unable to complete runtime initialization operations due to:
Azure.DataApiBuilder.Service.Exceptions.DataApiBuilderException: Cannot obtain Schema for entity customers with underlying database object source: dbo.customers due to: Login failed for user 'btspendo'.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by abhishekkumams**
## Why make this change?

- Closes #1321
  - special character like '&' is escaped during serialization.

## What is this change?

- Updated the Encoder used in the config serialization.
- Now using [`UnsafeRelaxedJsonEscaping`](https://learn.microsoft.com/en-us/dotnet/api/system.text.encodings.web.javascriptencoder.unsaferelaxedjsonescaping?view=net-7.0#system-text-encodings-web-javascriptencoder-unsaferelaxedjsonescaping), it does not escape HTML-sensitive characters such as <, >, &. Since we do not want any kind of change during string serialization, it's the best we can use.
- It also allows [UnicodeRanges.All](https://learn.microsoft.com/en-us/dotnet/api/system.text.unicode.unicoderanges.all?view=net-7.0#system-text-unicode-unicoderanges-all) to go through unescaped. Hence, it doesn't regress our existing changes.

## How was this tested?
- [x] Unit Tests
- I created a new user in Azure SQL with a password containing special characters such as [!@#$%^&*() ], and i was able to start the engine succesfully.

## NOTE:
Since the config file is json and it is storing connection-string as string,there are still some characters that can't be used like single quotes `'` or double quotes `"` , or a backslash `\` as password.

### CODE REVIEW

**Review by seantleonard — DISMISSED** [review]
Few questions

File: src/Cli/Utils.cs [hunk 1]
```diff
@@ -259,14 +258,15 @@ public class LowerCaseNamingPolicy : JsonNamingPolicy
...
+        /// Not escaping any special unicode characters.
         /// Ignoring properties with null values.
         /// Keeping all the keys in lowercase.
         /// </summary>
         public static JsonSerializerOptions GetSerializationOptions()
         {
             JsonSerializerOptions? options = new()
             {
-                Encoder = JavaScriptEncoder.Create(UnicodeRanges.All),
+                Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
```
[the remark this answers is not in the document]
Comment (seantleonard) [reply src/Cli/Utils.cs:269]: can you add this to pr description as examples of passwords which will not work?

Comment (Aniruddh25) [inline src/Cli/Utils.cs:269]: The docs say: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/character-encoding#serialize-all-characters to use this with caution. 
Can we use `SerializeToUtf8Bytes` so we can use this encoder safely?

Comment (abhishekkumams) [reply src/Cli/Utils.cs:271]: it creates a Byte array and then we would have convert it again to string. Furthermore, Under the caution, it is mentioned `Never allow the raw UnsafeRelaxedJsonEscaping output to be emitted into an HTML page or a <script> element`. Our usage doesn't violate this rule, instead it's directly consumed as a file. Secondly, this serialization option is in CLI only and used while generating the config. As CLi is meant for local development. this should be fine, I believe.

Comment (abhishekkumams) [reply src/Cli/Utils.cs:271]: we already use WriteAllText for generating the config: [here](https://github.com/Azure/data-api-builder/blob/ee22ea3a46f8b983ffe756a53b5e87bef022382b/src/Cli/Utils.cs#L877), which uses UTF-8 for default encoding as pointed out [here](https://github.com/Azure/data-api-builder/pull/1210/files#r1139458200) in the discussion.

**Review by Aniruddh25 — APPROVED** [review]

**Review by ayush3797 — APPROVED** [review]

**Pull request closed without merging**